### ISSUE TITLE
[Design] Notes

### ISSUE THREAD

**Issue by lpatmo** [codebuddies/v3#7]
Discussion about the concept of "notes" (or "learnings")

**Comment by lpatmo** [comment]
From @Grimmaldi:  

I'm not able to help write the code, but I wanted to try to clear up what I'd had in mind when I was discussing the site with @linda (she/her) and @wuworkshop yesterday.
As a user of Code Buddies, I might feel self-conscious about the Notes that I take.  Maybe I'm worried that I've misunderstood a concept or I don't feel it's up to the scrutiny of professionals or the world at large.  For that reason and a few other cases, I would want the option to make my Group and the Notes it contains visible only to members of that Group.
However, I agree with Bethany that Resources should always be shared and public.  This would allow the site to build up the tagged recommendation system of Resources that would still allow non-group members to follow either alone or with a group of their choosing.  As @bethanyg said, this approach would also cut down on duplication of entries in the Resources table (though, since typos and different naming conventions do occur, duplication would likely still exist).
The "hearts" on a Resource should be anonymous and each user should only be able to apply one to a resource, like a vote.  But, in my opinion, only public Notes about a resource should appear within the Resource.  A private Group's notes about a resource should only be visible to members of that group.
That's what I understood from our discussion, but, obviously, this is only my point of view.

**Comment by lpatmo** [comment]
@Grimmaldi I am guessing you'd want these notes to support a WYSIWYG, right? And code formatting?

**Comment by Grimmaldi** [comment]
I think code formatting would be an important feature for a site devoted to programming, but what do you mean by a WYSIWYG?   I mean, I understand that stands for "What You See Is What You Get", but I'm not sure how you that applies to Code Buddies notes.  Do you mean something similar to the Preview button that appears here on GitHub?  If so, that would also be a great feature.  If not, do you mind explaining what you had in mind?

**Comment by lpatmo** [comment]
Yes, by WYSIWYG I meant something like this:

![image](https://user-images.githubusercontent.com/4512699/67061622-b1bbe880-f115-11e9-8355-ba2ae59f8282.png)

I think classically they're called WYSIWYG editors :)

**Comment by Grimmaldi** [comment]
Yes, that would be nice, particularly for users who aren't familiar with markdown. :)
